tests/core/usage_statistics/test_execution_environment.py

```diff
@@ -0,0 +1,105 @@
+from typing import List
+from unittest import mock
+
+from packaging import version
+
+from great_expectations.core.usage_statistics.execution_environment import (
+    GEExecutionEnvironment,
+    InstallEnvironment,
+    PackageInfo,
+)
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.execution_environment.GEExecutionEnvironment._get_all_installed_packages",
+    return_value=True,
+)
+@mock.patch("importlib.metadata.version", return_value=True)
+@mock.patch(
+    "great_expectations.core.usage_statistics.package_dependencies.GEDependencies.get_dev_dependency_names",
+    return_value=True,
+)
+@mock.patch(
+    "great_expectations.core.usage_statistics.package_dependencies.GEDependencies.get_required_dependency_names",
+    return_value=True,
+)
+def test_get_installed_packages(
+    get_required_dependency_names,
+    get_dev_dependency_names,
+    mock_version,
+    get_all_installed_packages,
+):
+    """Test that we are able to retrieve installed and not installed packages in the GE execution environment."""
+
+    get_required_dependency_names.return_value = [
+        "req-package-1",
+        "req-package-2",
+        "not-installed-req-package-1",
+        "not-installed-req-package-2",
+    ]
+    get_dev_dependency_names.return_value = [
+        "dev-package-1",
+        "dev-package-2",
+        "not-installed-dev-package-1",
+        "not-installed-dev-package-2",
+    ]
+    mock_version.return_value = "8.8.8"
+    get_all_installed_packages.return_value = [
+        "req-package-1",
+        "req-package-2",
+        "dev-package-1",
+        "dev-package-2",
+    ]
+
+    ge_execution_environment: GEExecutionEnvironment = GEExecutionEnvironment()
+    expected_dependencies: List[PackageInfo] = [
+        PackageInfo(
+            package_name="req-package-1",
+            installed=True,
+            install_environment=InstallEnvironment.REQUIRED,
+            version=version.Version("8.8.8"),
```

Comment:
Are we sure this works again Git hashes or other edge cases like `-e` installation? Could we perhaps add some items to test that?

Reply:
version.parse() should handle those cases, but good call on adding some test cases.

Reply:
Added!